c/shared/source/ufowrite/ufowrite.c

```diff
@@ -650,6 +553,219 @@ static void writeBlueValues(ufwCtx h, abfPrivateDict *privateDict){
     }
 }
 
+static void writeFDArray(ufwCtx h, abfTopDict *top, char *buffer) {
+    abfPrivateDict *privateDict;
+    int i;
+    int j;
+    writeLine(h, "\t<key>com.adobe.type.FSType</key>");
+    sprintf(buffer, "\t<integer>%d</integer>", (int)h->top->FSType);
+    writeLine(h, buffer);
+    writeLine(h, "\t<key>com.adobe.type.postscriptFDArray</key>");
+    writeLine(h, "\t<array>");
+    for (j = 0; j < top->FDArray.cnt; j++) {
+        writeLine(h, "\t<dict>");
+        abfFontDict *fd = &h->top->FDArray.array[j];
+        writeLine(h, "\t<key>FontName</key>");
+        sprintf(buffer, "\t<string>%s</string>", fd->FontName.ptr);
+        writeLine(h, buffer);
+        writeLine(h, "\t<key>PaintType</key>");
+        sprintf(buffer, "\t<integer>%ld</integer>", fd->PaintType);
+        writeLine(h, buffer);
+        if (fd->FontMatrix.cnt == ABF_EMPTY_ARRAY) {
+            fd->FontMatrix.cnt = 6;
+            fd->FontMatrix.array[0] = 0.001;
+            fd->FontMatrix.array[1] = 0.0;
+            fd->FontMatrix.array[2] = 0.0;
+            fd->FontMatrix.array[3] = 0.001;
+            fd->FontMatrix.array[4] = 0.0;
+            fd->FontMatrix.array[5] = 0.0;
+        }
+        writeLine(h, "\t<key>FontMatrix</key>");
+        writeLine(h, "\t<array>");
+        for (i = 0; i < fd->FontMatrix.cnt; i++) {
+            float stem = fd->FontMatrix.array[i];
+            if (stem == ((int)stem))
+                sprintf(buffer, "\t\t<integer>%d</integer>", (int)stem);
+            else
+                sprintf(buffer, "\t\t<real>%.3f</real>", stem);
+            writeLine(h, buffer);
+        }
+        writeLine(h, "\t</array>");
+
+        privateDict = &(fd->Private);
+        writeLine(h, "\t<key>PrivateDict</key>");
+        writeLine(h, "\t<dict>");
+        writeBlueValues(h, privateDict);
+        writeLine(h, "\t</dict>");
+        writeLine(h, "\t</dict>");
+    }
+    writeLine(h, "\t</array>");
+}
+
+static void writeCIDMap(ufwCtx h, abfTopDict *top, char *buffer) {
+    int i;
+    writeLine(h, "\t<key>com.adobe.type.postscriptCIDMap</key>");
+    writeLine(h, "\t<dict>");
+    for (i = 0; i < h->glyphs.cnt; i++) {
+        sprintf(buffer, "\t\t<key>%s</key>", h->glyphs.array[i].glyphName);
+        writeLine(h, buffer);
+        sprintf(buffer, "\t\t<integer>%d</integer>", h->glyphs.array[i].cid);
+        writeLine(h, buffer);
+    }
+    writeLine(h, "\t</dict>");
+}
+
+static void writeGlyphOrder(ufwCtx h) {
```

Comment:
Seems like this does more than just write the glyph order.

Reply:
renamed the parse & write funcs in uforead & ufowrite from GlyphOrder -> LibPlist